yt_dlp/extractor/whyp.py

```diff
@@ -0,0 +1,45 @@
+from .common import InfoExtractor
+from ..utils import traverse_obj
+
+
+class WhypIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?whyp\.it/tracks/(?P<number_id>[0-9_-]+)/(?P<display_id>[a-zA-Z0-9_-]+)'
+    _TEST = {
+        'url': 'https://www.whyp.it/tracks/18337/home-page-example-track-b4kq7',
+        'md5': 'c1187b42ebf8605284e3dc92aeb33d16',
+        'info_dict': {
+            'url': 'https://cdn.whyp.it/50eb17cc-e9ff-4e18-b89b-dc9206a95cb1.mp3',
+            'id': '18337',
+            'title': 'Home Page Example Track',
+            'description': 'Just an example track for the home page!\n\nCredit: https://file-examples.com/index.php/sample-audio-files/sample-mp3-download/',
+            'ext': 'mp3',
+            'duration': 52.82,
+            'uploader': 'Brad',
+            'uploader_id': 1,
+            'thumbnail': 'https://cdn.whyp.it/a537bb36-3373-4c61-96c8-27fc1b2f427a.jpg',
+            'vcodec': 'none',
+        }
+    }
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        unique_id = mobj.group('number_id')
+
+        webpage = self._download_webpage(url, unique_id)
+
+        data = self._search_nuxt_data(webpage, unique_id)['rawTrack']
+
+        return {
+            'url': data['audio_url'],
+            'id': unique_id,
+            'title': data.get('title'),
+            'description': data.get('description'),
+            'duration': data.get('duration'),
+            'uploader': traverse_obj(data, ('user', 'username')),
+            'uploader_id': traverse_obj(data, ('user', 'id')),
+            'thumbnail': data.get('artwork_url'),
```

Comment:
Dict key traversal with `traverse_obj` can make the metadata extraction a bit less grueling.
`traverse_obj(data, {})` will always return a dict, so it is safe to unpack in return

It's a good idea to always call `float_or_none` or `int_or_none` on the duration value, in case it was a numeric string or otherwise invalid in the site's JSON. It's done below using a `traverse_obj` transformation, e.g. `{float_or_none}`

`uploader_id` should be a string, but is an integer in the JSON, so we want to safely stringify it with `str_or_none`. (you will need to update the test)
```suggestion
            **traverse_obj(data, {
                'title': 'title',
                'description': 'description',
                'duration': ('duration', {float_or_none}),
                'uploader': ('user', 'username'),
                'uploader_id': ('user', 'id', {str_or_none}),
                'thumbnail': ('artwork_url', {url_or_none}),
            }),
```
import `url_or_none`, `float_or_none` and `str_or_none` from utils